src/utils/function/sleep.ts

```
@@ -0,0 +1,16 @@
+// src/utils/function/sleep.ts
+
+/**
+ * Pauses execution for a specified number of milliseconds.
+ *
+ * @param ms - The number of milliseconds to sleep.
+ * @returns A promise that resolves after the specified delay.
+ * @throws Error if the input is not a valid positive number.
+ */
+export const sleep = (ms: number): Promise<void> => {
+    if (typeof ms !== 'number' || isNaN(ms) || ms < 0) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)